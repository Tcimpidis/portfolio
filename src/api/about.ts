export interface AboutMeDataDto {
  story: string;
  passion: string;
}