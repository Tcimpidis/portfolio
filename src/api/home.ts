import { CompanyType } from "./portfolio";

export interface HomeDataDto {
  summary: string;
  highlightReel: Highlight[];
}

export interface Highlight {
  imageName: string;
  company: CompanyType;
  title: string;
}

export const HomeData: HomeDataDto = {
  summary: `I am a full stack developer with over **8 years** of developing production ready, globally distributed software. I've prodomantently worked for logistics companies, with a focus on customer facing **SaaS web applications**. My passion being towards the frontend, I've strived to work and that domain. I have over 5 years experience working with **React** and **Typspcript**. Recently I've taken on learning **Next.js** to leverage to full capacity of **React 18**.`,
  highlightReel: [
    {
      imageName: 'chrobinson',
      company: "C.H Robinson worldwide",
      title: "Software Engineer III",
    },
    {
      imageName: 'delhivery',
      company: "Delhivery",
      title: "Senior Software Development Engineer",
    },
    {
      imageName: 'tcimpidis',
      company: "Tcimpidis",
      title: "Creator",
    },
  ]
}