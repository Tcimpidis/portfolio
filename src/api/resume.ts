export interface ResumeDataDto {
  summary: string;
  experience: ResumeDetailDto[];
  education: ResumeDetailDto[];
  extracurricular: ResumeDetailDto[];
}

export interface ResumeDetailDto {
 entity: string;
 location: string;
 title: string;
 duration: string;
 responsibilities: string[]; 
}

export const ResumeData:ResumeDataDto = {
  summary: "Senior Software Development Engineer. 8+ years experience specializing in full-stack development with a focus micro-service/micro-frontend architecture design. Passionate about creating robust scalable solutions using the newest technologies.",
  experience: [{
    entity: "Delhivery USA LLC",
    location: "Los Angeles, CA",
    title: "SENIOR SOFTWARE DEVELOPMENT ENGINEER",
    duration: "April 2022 - December 2023",
    responsibilities: [
      "Serve as architect for platform frontend.",
      "Oversee a team of three developers.",
      "Build deployment pipelines in github.",
      "Configure and manage cloudfront CDN and S3 bucket permissions in AWS.",
      "Design, implement and publish scalable component libraries.",
      "Work closely with stakeholders to build technical requirement documents.",
      "Estimate effort and create execution plans for deliverables.",
      "Define coding standards.",
      "Ensure code quality via code reviews. peer feedback sessions, and github action quality gates.",
      "Advise on conforming to security best practices.",
    ], 
  },
  {
    entity: "C.H.Robinson Worldwide",
    location: "Eden Prairie, MN",
    title: "SOFTWARE ENGINEER III",
    duration: "January 2015 - April 2022",
    responsibilities: ["A lead developer for C.H.Robinson's smallbusiness application https://Freightquote.com/book.",
    "Design and implement kafka event streams.",
    "Design and implement data management strategies through building data pipelines, which put data into UI models before consumption.",
    "Ensuring on demand data integrity.",
    "design and implement microservices.",
    "Website design and implementation.",
    "Collaborate with UX to build functional, scalable and responsive components.",
    "Architect redux state management.",
    "Work closely with security to ensure routes and applications aren't easy subjects for malicious attacks.",
    "Create event-driven architecture diagrams and flowcharts.",
    "Decompose existing site into micro-sites.",
    "Perform in-depth code reviews.","Work across various teams and technologies to ensure business needs are met and delivered.",
    "Engaged in sprint planning and grooming.",
    "Mentored new hires.",
    "Developed solutions for large-scale business initiatives.",
    "Designed and developed coding standards and practices, enforced them with in our team.",
    "Created scalable, clean code that passes quality gates.",
    "Triage support/production issues.",
    "Work directly with business shareholders.","Built code enhancements.",
    "Performed bug analysis.",
    "Wrote sql queries to pull data.",
   "Owned and managed deployments."], 
  }],
  education: [{
    entity: "Concordia University",
    location: "St Paul, MN",
    title: "CERTIFICATE IN SOFTWARE DEVELOPMENT",
    duration: "March 2015 - July 2015",
    responsibilities: [
      "Hands-on software developer apprenticeship with full-stack experience.", 
      "C# language fundamentals: variable declaration, looping constructs, conditional statements, class modeling, exception handling, LINQ Queries, unit testing and n-Tier application structure",
      "Web development fundamentals: ASP.Net MVC5, HTML5, CSS3, JavaScript, jQuery and AngularJS",
      "Database fundamentals: SQL server 2014, T-SQL query writing, stored procedures, data modeling and table creation"
    ], 
  },
  {
    entity: "Santa Fe College",
    location: "Gainesville, FL",
    title: "A.S. BOTANY",
    duration: "August 2007 - May 2010",
    responsibilities: ["Graduated with honors.", "Cumulative GPA:3.57."], 
  }],
  extracurricular: [{
    entity: "C.H.Robinson Worldwide - Robinson Cares",
    location: "Eden Prairie, MN",
    title: "TECHNICAL LEAD",
    duration: "January 2016 - April 2022",
    responsibilities: [
    "Co-lead all technical aspects of defining, creating and implementing the internal auction site for Robinson Cares charity group.",
    "Built out inital implementation in AngularJS",
    "Uplift site to use newer company tech-stack including setting up new devops workflows.",
    "Owned implementation of newer Oauth workflow to use windows authentication to automatically retreive and use JWT for signed in user.", 
    "Owned deployments.",
    "Triage production issue amd apply fixes. Sometimes while auction was active."], 
  }],
}