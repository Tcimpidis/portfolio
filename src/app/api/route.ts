import { ContactMeEmailTemplate } from '../../common/email/contact-me';
import { Resend } from 'resend';

const resend = new Resend(process.env.RESEND_API_KEY);

export async function POST(request: Request) {
  const { fullname, message, email, subject } = await request.json() as any; 
  try {
    const data = await resend.emails.send({
      from: `${process.env.ADMIN_RECIPIENT_EMAIL}`,
      to: [`${process.env.ADMIN_MAILTO_EMAIL}`],
      subject: subject,
      react: ContactMeEmailTemplate({ fullName:fullname , email, message }),
      text: ''
    });

    return Response.json(data);
  } catch (error) {
    return Response.json({ error });
  }
}