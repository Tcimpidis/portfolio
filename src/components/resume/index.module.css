.container {  
  display: flex;
  flex-direction: column;
}

.header{ 
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.location {
  composes: label from '../../app/layout.module.css';
  color: var(--tcimpidis-cool-shower);
  opacity: 1;
}

.duration {
  composes: label from '../../app/layout.module.css';
  color: var(--tcimpidis-dusk-on-the-lake);
  font-style: italic;
  font-weight: 400;
}

.entity {
  composes: label from '../../app/layout.module.css';
  color: var(--tcimpidis-dusk-on-the-lake);
}

.title {
  composes: label from '../../app/layout.module.css';
  color: var(--tcimpidis-dusk-on-the-lake);
  font-weight: 500;
}

.detail {
  composes: detail from '../../app/layout.module.css';
}
