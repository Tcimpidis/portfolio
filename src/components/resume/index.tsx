import styles from './index.module.css';

export const ResumeComponent = () => (
  <div className={styles.container}>
    <div className={styles.heading}>
      Jamin Tcimpidis
    </div>
    <div className={styles.contact}>
      <span></span>
    </div>
  </div>
)