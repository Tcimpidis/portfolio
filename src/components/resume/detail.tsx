import { ResumeDetailDto } from '@/api/resume'
import styles from './detail.module.css'
import { FC } from 'react';

interface ResumeDetailProps {
  detail: ResumeDetailDto;
}


export const ResumeDetail: FC<ResumeDetailProps> = (
  {detail:{
    entity,
    location,
    title,
    duration,
    responsibilities
  }}
) => (
  <div className={styles.container}>
    <div className={styles.header}>
      <span className={styles.entity}>{entity}</span>
      <span className={styles.location}>{location}</span>
    </div>
    <div className={styles.header}>
      <span className={styles.title}>{title}</span>
      <span className={styles.duration}>{duration}</span>
    </div>
    <ul className={styles.detail}>
      {responsibilities.map((value, i) => <li key={i}>{value}</li>)}
    </ul> 
  </div>
)