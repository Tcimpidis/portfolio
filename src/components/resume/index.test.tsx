import { ResumeData } from "@/api/resume";
import { render } from "@testing-library/react";
import { ResumeDetail } from ".";

describe(ResumeDetail.name, () => {
  const { education } = ResumeData;
  const props = {
    detail: education[0],
  };
  it("snapshot", () => {
    const { asFragment } = render(<ResumeDetail {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
