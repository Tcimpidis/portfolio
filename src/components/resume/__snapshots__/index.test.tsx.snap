// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResumeDetail snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <span
        class="entity"
      >
        Concordia University
      </span>
      <span
        class="location"
      >
        St Paul, MN
      </span>
    </div>
    <div
      class="header"
    >
      <span
        class="title"
      >
        CERTIFICATE IN SOFTWARE DEVELOPMENT
      </span>
      <span
        class="duration"
      />
    </div>
    <ul
      class="detail"
    >
      <li>
        Hands-on software developer apprenticeship with full-stack experience.
      </li>
      <li>
        C# language fundamentals: variable declaration, looping constructs, conditional statements, class modeling, exception handling, LINQ Queries, unit testing and n-Tier application structure
      </li>
      <li>
        Web development fundamentals: ASP.Net MVC5, HTML5, CSS3, JavaScript, jQuery and AngularJS
      </li>
      <li>
        Database fundamentals: SQL server 2014, T-SQL query writing, stored procedures, data modeling and table creation
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
